threescaleAPI-operator/pkg/threescaleAPI/handler.go

```diff
@@ -53,50 +53,70 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		accessToken := o.Spec.The3ScaleConfig.AccessToken
 		serviceName := o.Name
 
-		service, err := ensureServiceExists(c, accessToken, serviceName)
-		if err != nil {
-			return fmt.Errorf("error synchronizing service state - %s", err.Error())
+		if event.Deleted {
 
-		}
+			fmt.Printf("[-] Deleting service: %s\n", serviceName)
 
-		existingEndpoints, err := getEndpointsFrom3scaleSystem(c, accessToken, service)
-		if err != nil {
-			fmt.Printf("Couldn't get Endpoints from 3scale: %v\n", err)
+			service, err := getServiceFromServiceSystemName(c, accessToken, serviceName)
+			if err != nil {
+				return fmt.Errorf("Service %s doesn't exists, error: %s", serviceName, err.Error())
```

Comment:
I think this logic is slightly flawed because if an object doesnt exist (404) we get an error from `getServiceFromServiceSystemName` function. In this case, I dont think we should be reporting an error here, instead, we should just return?